flocker/node/agents/_logging.py

```diff
@@ -94,3 +99,70 @@
     [VOLUME_ID, STATUS, TARGET_STATUS, WAIT_TIME],
     u"Waiting for a volume to reach target status.",)
 # End: Helper datastructures used by AWS storage driver.
+
+# Begin: Helper datastructures used by OpenStack storage drivers
+
+CODE = Field.for_types("code", [int], u"The HTTP response code.")
+MESSAGE = Field.for_types(
+    "message", [bytes, unicode],
+    u"A human-readable error message given by the response.",
+)
+DETAILS = Field.for_types("details", [dict], u"Extra details about the error.")
+REQUEST_ID = Field.for_types(
+    "request_id", [bytes, unicode],
+    u"The unique identifier assigned by the server for this request.",
+)
+URL = Field.for_types("url", [bytes, unicode], u"The request URL.")
+METHOD = Field.for_types("method", [bytes, unicode], u"The request method.")
```

Comment:
Actually... Some of this looks familiar.  We have some HTTP request-related Eliot logging stuff over in `flocker/restapi/_logging.py`.

It would be nice to re-use this stuff instead of re-defining the fields.  I'm not exactly sure how best to do this.  Maybe this merits another module in `flocker.common`?  I'm not sure.  Maybe @itamarst will weigh in.